File: src/001.Array/053.MaximumSubarray/53.最大子数组和.go
/*
 * @lc app=leetcode.cn id=53 lang=golang
 *
 * [53] 最大子数组和
 */

// @lc code=start
func maxSubArray(nums []int) int {
	max := nums[0]
	sum := nums[0]

	for i := 1; i < len(nums); i++ {
		if sum + nums[i] < nums[i] {
			sum = nums[i]

			if sum > max {
				max = sum
			}

			continue
		}

		sum += nums[i]
	
		if sum > max {
			max = sum
		}
	}

	return max
}
// @lc code=end

File: src/001.Array/053.MaximumSubarray/53.最大子数组和.js
/*
 * @lc app=leetcode.cn id=53 lang=javascript
 *
 * [53] 最大子数组和
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
  let max = nums[0];
  let sum = nums[0];
  
  for (let i = 1; i < nums.length; i++) {
    if (sum + nums[i] < nums[i]) {
      sum = nums[i];

      if (max < sum) {
        max = sum;
      }

      continue;
    }

    sum = sum + nums[i];

    if (max < sum) {
      max = sum;
    }
  }
  
  return max;
};
// @lc code=end

